src/storage-client/src/storage_collections.rs

```diff
@@ -2286,15 +2286,20 @@ where
             >,
         > = FuturesUnordered::new();
 
-        let gen_upper_future = |id, mut handle: WriteHandle<_, _, _, _>, prev_upper| {
-            let fut = async move {
-                handle.wait_for_upper_past(&prev_upper).await;
-                let new_upper = handle.shared_upper();
-                (id, handle, new_upper)
-            };
+        let gen_upper_future =
+            |id, mut handle: WriteHandle<_, _, _, _>, prev_upper: Antichain<T>| {
+                let fut = async move {
+                    assert!(
```

Comment:
I'd prefer a soft assert! Don't want to panic envd if we missed something, and leaking a task is a very minor issue in comparison.